pkg/ingester/ingester_v2.go

```diff
@@ -93,6 +97,16 @@ func (u *userTSDB) Compact() error {
 
 // compactHead compacts the Head block at specified block durations avoiding a single huge block.
 func (u *userTSDB) compactHead(blockDuration int64) error {
+	u.forcedCompactionInProgressMtx.Lock()
+	u.forcedCompactionInProgress = true
+	u.forcedCompactionInProgressMtx.Unlock()
+	defer func() {
+		u.forcedCompactionInProgress = false
```

Comment:
You need to lock around this

Reply:
Instead of lock, what about using an atomic variable? Both would be fine actually.

Reply:
Added the lock